api/src/main/java/io/opencensus/metrics/DoubleGaugeMetric.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.metrics;
+
+import io.opencensus.common.ToDoubleFunction;
+import java.util.List;
+
+// TODO(mayurkale) : Add comments and examples
+public abstract class DoubleGaugeMetric<T> implements Gauge {
+
+  public abstract DataPoint<T> addDataPoint(List<LabelValue> labelValues);
+
+  public abstract DataPoint<T> addDataPoint(
```

Comment:
Don't need to return a DataPoint here correct? user can do whatever they want (reset, inc, etc.) via the combination {obj, function}

Reply:
Make sense